bytes_ostream.hh

```diff
@@ -460,4 +461,26 @@ public:
             _begin->next.reset();
         }
     }
+
+    // Makes this instance empty using async continuations, while allowing yielding.
+    //
+    // The first buffer is not deallocated, so callers may rely on the
+    // fact that if they write less than the initial chunk size between
+    // the clear() calls then writes will not involve any memory allocations,
+    // except for the first write made on this instance.
+    future<> clear_gently() noexcept {
+        if (!_begin) {
+            co_return;
+        }
+        _begin->offset = 0;
+        _size = 0;
+        _current = _begin.get();
+        auto p = std::move(_begin->next);
+        while (p) {
+            // Avoid recursion when freeing chunks
+            auto p_next = std::move(p->next);
+            p = std::move(p_next);
```

Comment:
Can you explain more how the recursion would happen and why the extra p_next is needed?

Reply:
Yeah, the comment was copied from `~chunk`.

If this would have been coded simply as `p = std::move(p->next)` we need to remember that `std::move(p->next)` is not a function call (maybe we can wrap it a function to simply stuff).
Then, If `std::unique_ptr<T>& operator=(std::unique_ptr<T>&&)` first destroys `*this` and then move-constructs it in-place, the destructor will recursively destroy `*p` and `p->next` with it before performing the move.

Reply:
added elaborate comment in v3